invenio_rdm_records/services/components/pids.py

```diff
@@ -20,42 +20,48 @@
 class ExternalPIDsComponent(ServiceComponent):
     """Service component for pids."""
 
-    def _validate_pid(self, scheme, pid):
+    def _validate_pid(self, scheme, pid, record, provider=None):
         """Call provider to validate the given PID."""
-        provider_name = pid.get("provider")
-        client = pid.get("client")
-        provider = self.service.get_provider(scheme, provider_name, client)
-        if provider:
-            success, errors = provider.validate(**pid)
-            if errors:
-                raise ValidationError(message=errors, field_name="pids")
-
-    def _validate_pids(self, pids):
+        if not provider:  # In case we do not need to calculate it again
+            provider_name = pid.get("provider")
+            client = pid.get("client")
+            provider = self.service.get_provider(scheme, provider_name, client)
+            if not provider:
+                raise ValidationError(
+                    message=_(f"Provider {provider_name} not found for PID " +
+                              f"type {scheme}"),
+                    field_name="pids",
+                )
+
+        # NOTE: provider should not be None by now, if not configured should
```

Comment:
mine again 😅 we shouldn't use `NOTE` since this has meaning only for translations 👼